app/models/user.rb

```diff
@@ -32,14 +40,4 @@ def full_name
     return username unless first_name.present?
     "#{first_name} #{last_name}"
   end
-
-  protected
-
-  def password_required?
-    !password.nil? || !password_confirmation.nil?
-  end
-
-  def email_required?
```

Comment:
Some facebook users don't have email(due facebook permissions or the facebook email isn't comfirmed yet), with that you can't register that users, if you want that you have to check the presence of email in https://github.com/toptierlabs/rails_api_base/blob/master/app/models/concerns/facebookeable.rb#L15.

Otherwise you can return `facebook_id.blank?` instead of `false`